build/fbcode_builder/manifests/cmake

```diff
@@ -11,30 +11,17 @@ cmake
 [dependencies]
 ninja
 
-[download.os=windows]
```

Comment:
I'm not properly awake yet, but I worry that these windows and darwin changes will break things for those platforms; in particular, Windows doesn't have a predictable `make` utility, so it is desirable to sidestep that headache if we can!

Reply:
Hey @wez, thank you for the review.
At least on darwin (big sur) I had no problems with my current setup, however I do not have a windows machine. Do you prefer that we leave the dowload options for windows and darwin as they were?

IMO we should leave darwin just as linux, since apple now supports ARM archs, which will eventually cause problems for some users (like me).

Reply:
But really sure if we can do conditionals at the manifest declaration, because if we can I can change in a way that it always downloads the binaries (for ARM/x86) for all platforms.

Reply:
I'm 50:50 on making darwin build it: darwin's arm platforms can run intel binaries, so the build is probably faster if we have it download the binary pre-built.

Windows I think is important to download binaries because there's no guarantee that there is a working make at all, let alone one that targets native windows binaries.

The `.os=windows` stuff at the end of the key is the getdeps conditional syntax.
You could add `arch` as a context variable over here:

https://github.com/facebook/openr/blob/f0e6d03851875b0c85b9e02c8a5ee7924f3d5920/build/fbcode_builder/getdeps/buildopts.py#L167

if you wanted to allow something like `[download.all(os=darwin, arch=arm)]`

Reply:
Awesome, ty for the input.
Do you prefer that we allow linux to download the ARM binaries instead of compiling it?

Reply:
if building it isn't dramatically slower than downloading it, then building seems fine on linux; generally nicer to have fewer conditions in the manifest!

Reply:
@wez ty for your help, I've sent a new version which only compiles cmake on linux. I took the approach for letting the manifest with fewer conditions.

It did not take much time to compile cmake for me. (And I was running on a VM with 1 processor)

However if you prefer that we expose the arch type for the manifest. I can quickly implement this feature.